packages/shared/lib/appSettings.ts

```diff
@@ -1,18 +1,45 @@
 import { persistent } from './helpers'
+
+/**
+ * The types of themes for the app
+ */
+export type AppTheme = 'light' | 'dark' | 'system'
+
 /**
  * App Settings
  */
 export interface AppSettings {
     deepLinking: boolean
     language: string,
+    theme: AppTheme,
     darkMode: boolean,
     notifications: boolean
 }
 
 export const appSettings = persistent<AppSettings>('settings', {
     deepLinking: false,
     language: 'en',
+    theme: 'light',
     darkMode: false,
     notifications: true
 })
 
+/**
+ * NOTE: This media query is performed only once to help make the UX
+ * consistent across OS platforms. To be specific, it ensures that users
+ * are required to restart Firefly for system theme changes to take effect.
+ * This is because the behaviour of this media query is different among platforms,
+ * e.g. queries on MacOS result in up-to-date information whereas Linux-based platforms
+ * result in stale information.
+ */
+const isSystemInDarkMode = window.matchMedia('(prefers-color-scheme: dark)').matches
```

Comment:
Would the Electron [`nativeTheme.shouldUseDarkColors`](https://github.com/electron/electron/blob/v9.4.4/docs/api/native-theme.md#nativethemeshouldusedarkcolors-readonly) property work better here?

Reply:
It probably would, but I've tried to figure out how to access that from the Electron app instance. Using `require('electron')` requires me to add `@types/node` as a dependency, and I can't `import { nativeTheme } from 'electron'` to resolve correctly. If we're okay with having a `require(...)` statement then I can make those changes.

Reply:
Your best bet might be to call it via IPC like we did for `getOS`:

https://github.com/iotaledger/firefly/blob/4053f14653ff59e2c0b3c3beab0be0c53328adf5/packages/desktop/electron/preload.js#L179

https://github.com/iotaledger/firefly/blob/d2dda8475b7c92519beeb40894b7723f4eb8787b/packages/desktop/electron/main.js#L393-L396

Then, you can call it like this:
https://github.com/iotaledger/firefly/blob/0c07be772e84c0827379085a5a6715553ae7b375/packages/shared/components/TitleBar.svelte#L26

Reply:
I would be good to do it this way, but we should make sure that this way is mobile-friendly somehow. @amadeu2 do you know if Capacitor has a similar way to access platform configuration info, specifically to find the system's theme?

Reply:
I think we have a lot of electron stuff that will need to refactor and abstract from for mobile, so I suggest being selfish to desktop for now and choosing the best way for desktop.

Reply:
Since the `nativeTheme` is on the `remote` object, which if we want to use it requires enabling the `remote` module. I think we decided not to do that right before I joined a few months ago.

Reply:
Right, but we can expose it via IPC so we don't need to use `remote`

Reply:
@rajivshah3 @maxwellmattryan Should we merge this and make an issue to refactor?

Reply:
Sounds fine to me 👍